Client/src/Pages/InfrastructureMonitors/CreateMonitor/index.jsx

```diff
@@ -0,0 +1,432 @@
+	const theme = useTheme();
+
+	const idMap = {
+		"notify-email-default": "notification-email",
+	};
+
+	const [errors, setErrors] = useState({});
+
+	const alertErrKeyLen = Object.keys(errors).filter((k) =>
+		k.startsWith(THRESHOLD_FIELD_PREFIX)
+	).length;
+
+	const handleCustomAlertCheckChange = (event) => {
+		const { value, id } = event.target;
+		setInfrastructureMonitor((prev) => {
+			const newState = {
+				[id]: prev[id] == undefined && value == "on" ? true : !prev[id],
+			};
+			return {
+				...prev,
+				...newState,
+				[THRESHOLD_FIELD_PREFIX + id]: newState[id]
+					? prev[THRESHOLD_FIELD_PREFIX + id]
+					: "",
+			};
+		});
+		// Remove the error if unchecked
+		setErrors((prev) => {
+			return buildErrors(prev, [THRESHOLD_FIELD_PREFIX + id]);
+		});
+	};
+
+	const handleBlur = (event, appenedID) => {
+		event.preventDefault();
+		const { value, id } = event.target;
+		if (id?.startsWith("notify-email-")) return;
+		const { error } = infrastractureMonitorValidation.validate(
+			{ [id ?? appenedID]: value },
+			{
+				abortEarly: false,
+			}
+		);
+		setErrors((prev) => {
+			return buildErrors(prev, id ?? appenedID, error);
+		});
+	};
+
+	const handleChange = (event, appendedId) => {
+		event.preventDefault();
+		const { value, id } = event.target;
+		let name = appendedId ?? idMap[id] ?? id;
+
+		if (name.includes("notification-")) {
+			name = name.replace("notification-", "");
+			let hasNotif = infrastructureMonitor.notifications.some(
+				(notification) => notification.type === name
+			);
+			setInfrastructureMonitor((prev) => {
+				const notifs = [...prev.notifications];
+				if (hasNotif) {
+					return {
+						...prev,
+						notifications: notifs.filter((notif) => notif.type !== name),
+					};
+				} else {
+					return {
+						...prev,
+						notifications: [
+							...notifs,
+							name === "email"
+								? { type: name, address: value }
+								: // TODO - phone number
+									{ type: name, phone: value },
+						],
+					};
+				}
+			});
+		} else {
+			setInfrastructureMonitor((prev) => ({
+				...prev,
+				[name]: value,
+			}));
+		}
+	};
+
+	const generatePayload = (form) => {
+		let thresholds = {};
+		Object.keys(form)
+			.filter((k) => k.startsWith(THRESHOLD_FIELD_PREFIX))
+			.map((k) => {
+				if (form[k]) thresholds[k] = form[k];
+				delete form[k];
+				delete form[k.substring(THRESHOLD_FIELD_PREFIX.length)];
+			});
+
+		form = {
+			...form,
+			description: form.name,
+			teamId: user.teamId,
+			userId: user._id,
+			type: "hardware",
+			notifications: infrastructureMonitor.notifications,
+			thresholds,
+		};
+		return form;
+	};
+	const handleCreateInfrastructureMonitor = async (event) => {
+		event.preventDefault();
+		let form = {
+			...infrastructureMonitor,
+			name:
+				infrastructureMonitor.name === ""
+					? infrastructureMonitor.url
+					: infrastructureMonitor.name,
+			interval: infrastructureMonitor.interval * MS_PER_MINUTE,
```

Comment:
This conversion isn't correct for any values in seconds. If you select "15s" you get 900000ms, which is 15 minutes.  Since the constant is MS_PER_MINUTE, seconds should first be converted to minutes before multiplying by MS_PER_MINUTE.  ie 15 seconds is 0.25 minutes

Reply:
updated, the frequency already in seconds, so I need to change to MS_PER_SECOND, done.
```
	//select values
	const frequencies = [
		{ _id: 15, name: "15 seconds" },
		{ _id: 30, name: "30 seconds" },
		{ _id: 60, name: "1 minute" },
		{ _id: 120, name: "2 minutes" },
		{ _id: 300, name: "5 minutes" },
		{ _id: 600, name: "10 minutes" },
	];
```

Reply:
@jennifer-gan we use milliseconds per m in minute everywhere else in the app, can you convert the seconds to minutes instead of changing the conversion factor so that we stay consistent?

Otherwise someone is going to be confused at some point and convert the wrong values. Thanks!